indexdata.go

```diff
@@ -94,32 +94,39 @@ type symbolData struct {
 	// symContent stores Symbol.Sym and Symbol.Parent.
 	// TODO we don't need to store Symbol.Sym.
 	symContent []byte
-	symIndex   []uint32
+	symIndex   []byte
 	// symKindContent is an enum of sym.Kind and sym.ParentKind
 	symKindContent []byte
 	symKindIndex   []uint32
-	//symMetadata is [4]uint32 Sym Kind Parent ParentKind
+	// symMetadata is [4]uint32 Sym Kind Parent ParentKind
 	symMetaData []byte
 }
 
+func (d *symbolData) parent(i uint32) []byte {
+	delta := binary.BigEndian.Uint32(d.symIndex)
+	start := binary.BigEndian.Uint32(d.symIndex[i*4:]) - delta
+	end := binary.BigEndian.Uint32(d.symIndex[(i+1)*4:]) - delta
```

Comment:
I don't really understand this delta part. Can you explain?

Reply:
`parent` is the lazy version of `relativeIndex`.
```go
// relativeIndex returns the relative offsets of the items (first
// element is 0), plus a final marking the end of the last item.
func (s *compoundSection) relativeIndex() []uint32 {
	ri := make([]uint32, 0, len(s.offsets)+1)
	for _, o := range s.offsets {
		ri = append(ri, o-s.offsets[0])
	}
	if len(s.offsets) > 0 {
		ri = append(ri, s.data.sz)
	}
	return ri
}
```
Zoekt stores the absolute offsets for each simple section, so subtracting the offset of the first element (delta) from each elements' offset will give us the relative offset.